extensions/arc/src/models/controllerModel.ts

```diff
@@ -60,11 +60,36 @@ export class ControllerModel {
 		};
 	}
 
+	/**
+	 * acquires a login session to this controller. If a login session is in progress against any controller
+	 * then this call waits for the previous login session to finish before acquiring the login session.
+	 * @param promptReconnect - if true then the user is prompted for connection information.
+	 */
+	public async acquireLogin(promptReconnect: boolean = false): Promise<void> {
+		//wait for any previous login sessions that might be in progress to finish
+		if (ControllerModel._loginSessionInProgress) {
+			await ControllerModel._loginSessionInProgress;
+		}
+		// create a new in progress promise object
+		ControllerModel._loginSessionInProgress = new Deferred<void>();
```

Comment:
If multiple callers are waiting on this then they'll all get notified and continue execution at once which is the exact problem you're trying to fix in the first place.

Reply:
hmmm - let me think more on that. Do you have a suggestion?

Reply:
I am looking into chaining all the pending callers so they go in a sequence